utilities/merge_operators/cassandra/merge_operator.h

```diff
@@ -0,0 +1,33 @@
+/**
+ * A MergeOperator for rocksdb that implements Cassandra row value merge.
+ * @author Chen Shen (chenshen@fb.com)
+ * Copyright 2017 Facebook
+ */
+
+#pragma once
+#include "rocksdb/merge_operator.h"
+#include "rocksdb/slice.h"
+
+namespace rocksdb {
+namespace cassandra {
+
+class CassandraValueMergeOperator : public MergeOperator {
+public:
+  static std::shared_ptr<MergeOperator> CreateSharedInstance();
+
+  virtual bool FullMerge(const Slice& key,
+                         const Slice* existing_value,
+                         const std::deque<std::string>& operand_list,
+                         std::string* new_value,
+                         Logger* logger) const override;
+
+  virtual bool PartialMerge(const Slice& key,
```

Comment:
Have you considered instead implementing `PartialMergeMulti`? It can be more efficient if you are able to merge more than two operators at once.